src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx

```diff
@@ -0,0 +1,242 @@
+	onEditRights: (info: ShareUserInfo, newRights: ShareUserInfoRights) => void;
+	onDeleteUser: (info: ShareUserInfo) => void;
+};
+
+const ShareWithColleagues: FC<ShareWithColleaguesProps & UserProps> = ({
+	users,
+	user,
+	onAddNewUser,
+	onDeleteUser,
+	onEditRights,
+}) => {
+	const { tText } = useTranslation();
+	const currentUser = users.find((u) => u.email === user?.mail) as ShareUserInfo;
+	const [isRightsDropdownOpen, setIsRightsDropdownOpen] = useState<boolean>(false);
+	const [newShareUser, setNewShareUser] = useState<Partial<ShareUserInfo>>({
+		email: undefined,
+		rights: undefined,
+	});
+	const [error, setError] = useState<string | null>(null);
+	const [isEditRightsModalOpen, setIsEditRightsModalOpen] = useState<boolean>(false);
+	const [toEditShareUser, setToEditShareUser] = useState<ShareUserInfo>();
+
+	const handleRightsButtonClicked = () => {
+		setIsRightsDropdownOpen(!isRightsDropdownOpen);
+	};
+
+	const handleAddNewUser = () => {
+		if (/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]+$/.test(newShareUser.email as string)) {
+			onAddNewUser(newShareUser);
+			setError(null);
+		} else {
+			setError(
+				tText(
+					'shared/components/share-with-colleagues/share-with-colleagues___email-is-geen-geldig-emailadres'
+				)
+			);
+		}
+	};
+
+	const handleEditUserRights = (user: ShareUserInfo) => {
+		setToEditShareUser(user);
+		setIsEditRightsModalOpen(true);
+	};
+
+	const handleConfirmEditUserRights = (right: ShareUserInfoRights) => {
+		onEditRights(toEditShareUser as ShareUserInfo, right);
+		setToEditShareUser(undefined);
+	};
+
+	const handleDeleteUser = (info: ShareUserInfo) => {
+		onDeleteUser(info);
+	};
+
+	const updateNewShareUserInfo = (value: Record<string, string>) => {
+		setNewShareUser({
+			...newShareUser,
+			...value,
+		});
+	};
+
+	const renderColleaguesInfoList = () => {
+		if (users.length > 0) {
+			return (
+				<ul className="c-colleagues-info-list">
+					{sortShareUsers(users).map((user, index) => (
+						<li
+							key={`colleagues-info-list-item-${index}`}
+							className="c-colleague-info-row"
+						>
+							<div className="c-colleague-info-row__avatar">
+								<Avatar
+									initials={
+										user.firstName && user.lastName
+											? user.firstName[0] + user.lastName[0]
+											: user.email.slice(0, 2).toUpperCase()
+									}
+									image={user.profileImage}
+								/>
+							</div>
+
+							<div className="c-colleague-info-row__info">
+								{(user.firstName || user.lastName) && (
+									<p>{`${user.firstName} ${user.lastName}`}</p>
+								)}
+
+								<p className="c-colleague-info-row__info__email">
+									{truncate(user.email, {
+										length: 32,
+										separator: ' ',
+									})}
+								</p>
+							</div>
+
+							<div className="c-colleague-info-row__rights">
+								<span>{shareUserRightToString(user.rights)}</span>
+
+								{currentUser.rights === ShareUserInfoRights.OWNER &&
+									currentUser.email !== user.email && (
```

Comment:
Extract this condition to a const with a descriptive name so it's more clear what it checks
ex. "isOwner"